src/librustc_typeck/check/_match.rs

```diff
@@ -38,12 +38,11 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                 self.write_ty(pat.id, expected);
             }
             PatKind::Lit(ref lt) => {
-                self.check_expr(&lt);
-                let expr_ty = self.expr_ty(&lt);
+                let expr_t = self.check_expr(&lt);
```

Comment:
`expr_ty` is still used in a few places. I can keep going trying to remove it entirely if you like. Otherwise I can rename the variable `ty`.